stempy/reader.cpp

```diff
@@ -23,6 +23,30 @@ using std::vector;
 
 namespace stempy {
 
+Header::Header(uint32_t frameWidth, uint32_t frameHeight,
+               uint32_t imageNumInBlock,uint32_t blockNumInFile, uint32_t scanWidth, 
+               uint32_t scanHeight, uint32_t startPosition, vector<uint32_t>& imageNumbers)
+{
+    this->frameWidth = frameWidth;
+    this->frameHeight=frameHeight;
+    this->imagesInBlock = imageNumInBlock;
+    this->scanHeight=scanHeight;
+    this->scanWidth=scanWidth;
+    //if there is no imageNumbers in original data set, reconstruct it
+    if(imageNumbers.size()!=0){
+      this->imageNumbers=imageNumbers;
+    }else{
+      //this reconstruction is only tested ok for big_stem_image.hdf5
```

Comment:
I still don't understand this comment. Are you saying that this block is specific to a particular dataset?